### ISSUE TITLE
CosmosDB Mongo API - StartsWith (regex) doesn't work with aggregation facets

### ISSUE THREAD

**Issue by IvaskevychYuriy** [Azure/azure-cosmos-dotnet-v3#2690]
Disclaimer, I'm not sure if this is the right place to file a bug report, please suggest other repo/move automatically if needed, sorry in advance.

**Describe the bug**
I'm using Azure CosmosDB with MongoDB API via C# Mongo Driver. I utilize aggregation facets as a common way to get rows + count in one query. Targeting MongoDB works fine obviously, the issue is when I try to use Cosmos - I get an exception 
>Command aggregate failed: Unsupported regex option: 115.

Aggregate contains `$match` filtering stage with `$regex` inside translated from C# `string.StartsWith()`. The weird thing is that it works fine without aggregation or with aggregation but not with facet inside aggregation. 

**To Reproduce**
```charp
using System.Linq;
using System.Security.Authentication;
using System.Threading.Tasks;
using MongoDB.Bson;
using MongoDB.Driver;

namespace MongoDriverCosmosTest
{
	class Program
	{
		static async Task Main(string[] args)
		{
			string conn = "<cosmos conn>";
			var settings = MongoClientSettings.FromUrl(new MongoUrl(conn));
			settings.SslSettings = new SslSettings { EnabledSslProtocols = SslProtocols.Tls12 };
			var mongoClient = new MongoClient(settings);

			var db = mongoClient.GetDatabase("TestCompat");
			var col = db.GetCollection<A>("a");

			// Use this to populate initial test data
			//await col.InsertManyAsync(new[]
			//{
			//	new A { Name = "abc" },
			//	new A { Name = "test 1" },
			//	new A { Name = "other one" },
			//	new A { Name = "tested 2 abc" },
			//});

			// (1) Works - regular query
			var res1 = await col.Find(x => x.Name.StartsWith("test")).ToListAsync();

			var dataStages = new[] { PipelineStageDefinitionBuilder.Match<A>(x => x.Name.StartsWith("test")) };
			var pipelineDefinition = PipelineDefinition<A, A>.Create(dataStages);
			// (2) Works - aggregation
			var res2 = await col.Aggregate(pipelineDefinition).ToListAsync();

			var dataFacet = AggregateFacet.Create("data", pipelineDefinition);
			// (3) Throws an exception - aggregation with facet
			var aggregation = await col.Aggregate()
				.Facet(dataFacet)
				.ToListAsync();

			var res3 = aggregation.First()
				.Facets.First(x => x.Name == dataFacet.Name)
				.Output<A>();

		}

		public class A
		{
			public ObjectId Id { get; set; }
			public string Name { get; set; }
		}
	}
}

```

Basically, the minimal example works fine in cases (1) and (2), but fails on (3) - when a facet is used in aggregation. In real world scenario there's another count facet that returns total count, but it can be omitted for simplicity here since this fails too.

Any ideas? Thanks in advance!

**Expected behavior**
Facet stages inside aggregation should work same way as just aggregation.

**Actual behavior**
An exception is thrown.

**Environment summary**
SDK Version:
OS Version: Windows 10
CosmosDB MongoDB API Version: `4.0`
MongoDB Driver: `2.11.3`, `2.13.1`(latest)
.Net Core: `3.1`

**Additional context**

**Comment by j82w** [comment]
This repo is for the .NET SQL API SDK. Please [contact support](https://aka.ms/azure-support) for help with the above issue.
